### PR TITLE
(3/n) update `UndoListing` to match new design for bulk collection actions

### PR DESCRIPTION
**Pull request by EmmadUsmani**
Epic: https://github.com/metabase/metabase/issues/28737

### Description

Third in a series of PR updating the bulk collection actions (move and archive) flow. ([Previous PR](https://github.com/metabase/metabase/pull/28797))

This PR centers the undo toast to match the position of the bulk actions toast.

### How to verify

1. Click on a collection in the sidebar.
2. Click on checkboxes to select items.
3. Either undo or archive.
4. Verify action took place and undo toast now appears in center.
5. Verify clicking undo reverts the action.

### Demo

https://user-images.githubusercontent.com/37751258/222258012-2c9ac52d-eb3b-4e79-913d-47d2da9ca4cb.mov

### Checklist

- [✅] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/App.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ import {
   NotFound,
   Unauthorized,
 } from "metabase/containers/ErrorPages";
-import UndoListing from "metabase/containers/UndoListing";
+import { UndoListing } from "metabase/containers/UndoListing";
```
Comment (EmmadUsmani) [inline frontend/src/metabase/App.tsx:13]: Start review here

Comment (EmmadUsmani) [reply frontend/src/metabase/App.tsx:13]: Updated `UndoListing` to be a named export, since they are preferred over default exports. See our [FE style guide notion doc.](https://www.notion.so/metabase/FE-Style-Guide-Proposals-Discussion-acdddeb36bd148e592d8be749ff9e4ef?pvs=4#01ed8c871d224d0daf5c45ee05e6b46e)

File: frontend/src/metabase/containers/UndoListing.jsx [hunk 2]
```diff
@@ -22,15 +23,6 @@ import {
   UndoList,
 } from "./UndoListing.styled";
 
-const mapStateToProps = (state, props) => ({
```
Comment (EmmadUsmani) [inline frontend/src/metabase/containers/UndoListing.jsx:25]: Removing `mapStateToProps` and `mapDispatchToProps` in favor of using the `useSelector` and `useDispatch` hooks, see below in component body.

File: frontend/src/metabase/containers/UndoListing.jsx [hunk 3]
```diff
@@ -62,49 +54,57 @@ UndoToast.propTypes = {
...
-        </CardContentSide>
-        <CardContentSide>
-          {undo.actions?.length > 0 && (
-            <UndoButton onClick={onUndo}>{t`Undo`}</UndoButton>
-          )}
-          <DismissIcon name="close" onClick={onDismiss} />
-        </CardContentSide>
-      </CardContent>
-    </ToastCard>
+    <Motion
```
Comment (EmmadUsmani) [inline frontend/src/metabase/containers/UndoListing.jsx:58]: Previously this component had no animation. This was fine in the old design, but in the new design the `BulkActions` toast component is animated while entering the screen. It would look jarring if it was then immediately replaced by the `UndoListing` toast without an entrance animation, so I added one.

File: frontend/src/metabase/containers/UndoListing.jsx [hunk 4]
```diff
@@ -62,49 +54,57 @@ UndoToast.propTypes = {
...
 }
 
-UndoListing.propTypes = {
-  undos: PropTypes.array.isRequired,
-  performUndo: PropTypes.func.isRequired,
-  dismissUndo: PropTypes.func.isRequired,
-};
+function UndoListingInner() {
+  const dispatch = useDispatch();
+  const undos = useSelector(state => state.undo);
```
Comment (EmmadUsmani) [inline frontend/src/metabase/containers/UndoListing.jsx:93]: I deleted the `getUndo` selected and `undo.js` since it seemed overkill to have an entire separate selector and file that was located far from the single component that uses it.

File: frontend/src/metabase/App.tsx (hunk 1 above)
Comment (kulyk) [reply frontend/src/metabase/App.tsx:13]: Actually, I'm not sure that's the case, most of Metabase exports are default 🤔 I think it was a proposal we haven't really implemented (cc @alxnddr)

File: e2e/test/scenarios/collections/collections.cy.spec.js [hunk 5]
```diff
@@ -472,6 +472,11 @@ describe("scenarios > collection defaults", () => {
           // Check that items were actually moved
           navigationSidebar().findByText("First collection").click();
           cy.findByText("Orders");
+
+          cy.findByText("Undo").click();
+          navigationSidebar().findByText("Our analytics").click();
+          cy.findByText("Orders").should("exist");
```
Comment (kulyk) [inline e2e/test/scenarios/collections/collections.cy.spec.js:478]: Maybe `should("be.visible")` would be more helpful?

File: frontend/src/metabase/containers/UndoListing.jsx [hunk 6]
```diff
@@ -62,49 +54,57 @@ UndoToast.propTypes = {
...
+      {({ translateY }) => (
+        <ToastCard
+          dark
+          data-testid="toast-undo"
+          style={{
+            ...{ transform: `translateY(${translateY}px)` },
+            ...(undo.toastColor
+              ? { backgroundColor: color(undo.toastColor) }
+              : {}),
+          }}
```
Comment (kulyk) [inline frontend/src/metabase/containers/UndoListing.jsx:71]: I personally find that kind of conditional spreading pretty hard to parse. Could we refactor it somehow?

File: frontend/src/metabase/containers/UndoListing.jsx [hunk 7]
```diff
@@ -62,49 +54,57 @@ UndoToast.propTypes = {
...
-};
+function UndoListingInner() {
+  const dispatch = useDispatch();
+  const undos = useSelector(state => state.undo);
+  const isNavbarOpen = useSelector(getIsNavbarOpen);
 
-function UndoListing({ undos, performUndo, dismissUndo }) {
   return (
-    <UndoList>
+    <UndoList style={{ marginLeft: isNavbarOpen ? NAV_SIDEBAR_WIDTH_HALF : 0 }}>
```
Comment (kulyk) [inline frontend/src/metabase/containers/UndoListing.jsx:97]: Should we move this styling into `UndoList`?

File: frontend/src/metabase/App.tsx (hunk 1 above)
Comment (alxnddr) [reply frontend/src/metabase/App.tsx:13]: That's true, the vast majority of engineers preferred using default exports. I'll drop this section from the doc

Comment (EmmadUsmani) [reply frontend/src/metabase/App.tsx:13]: FWIW I think it's worth at least keeping in the doc even if many engineers don't follow and we don't enforce it strictly. Perhaps in the future we can enforce it with a [lint rule](https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-default-export.md).

File: frontend/src/metabase/containers/UndoListing.jsx (hunk 6 above)
Comment (EmmadUsmani) [reply frontend/src/metabase/containers/UndoListing.jsx:71]: Yeah I can add props to the styled component

**Review by kulyk — APPROVED** [review]
🚢

**Review by alxnddr — APPROVED** [review]
